docs/proposals/odo-config-file.md

```diff
@@ -0,0 +1,132 @@
+
+```
+odo create wildfly example --memory=1GB
+
+# This should set value in .odo config file   (memory=2GB)
+```
+
+### Use configuration to "decorate" the deployment
+On component creation, via `odo create ...` the CLI will take information stored in this configuration file to create an actual deployment on OpenShift.
+
+```
+# Value in .odo config file   (memory=2GB)
+odo create wildfly example
+```
+
+The deployment that should be created, would have the following configuration:
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+spec:
+  containers:
+  - image: xxx
+    resources:
+      limits:
+        memory: 2Gi
+      requests:
+        memory: 2Gi
+```
+
+### Use configuration to "decorate" the CLI behavior
+On component creation, via `odo create ...` the CLI will take information stored in this configuration file to define some values that would typically be defined by the CLI and that are relative to how this works..
+
+```
+# Value in .odo config file   (app-name=myapp, component-name=mycomp)
+odo create wildfly
+# Would create an application called myapp and component named mycomp using wildfly
+```
+
+
+### Developer friendly flags
+Configuration options that will decorate the deployment should try to not use (as much as possible) terminology used in k8s. An example are memory resource requests and limits. Instead one could use terms like max-mem, min-mem and mem, that could work like this:
+
+* When `mem` is set, memory requests and limits are the same.
+* `min-mem` should be memory requests.
+* `max-mem` should be memory limits. 
+* `mem` is exclusive with `min-mem` or `max-mem`
+
+### Configuration precedence
+If the CLI provides some configuration values as flags for configuration that is store in the file, the ones provided explicitly from the user should take precedence.
+
+```
+# Value in .odo config file   (memory=2GB)
+odo create wildfly example --memory=1GB
+
+# The value that should be set for memory is 1 GB
+```
+
+### Explicitly modify the .odo config file
+There might be times that a developer would want to add/edit/remove values from the .odo configuration file. There should be a command to manage this file in the odo CLI. That command should be the same used to touch configuration of the global `.odo` config file located in the users's home directory.
+
+Currently that command is:
+
+```
+odo utils config [set|get|delete|add] KEY [VALUE] --local
+```
+
+Some examples of this command could be:
+
+```
+odo utils config set memory 2Gi --local      # Set's component memory to 2 GB
+odo utils config delete memory --local       # Removes configuration for memory
+odo utils config add odoignore .odo --local  # Adds a value to odoignore list
+odo utils config list --local                # List all config
+```
+
+## Proposed information to be stored
+We need to define and delimit what information will be stored in this file. 
+
+### Deployment decoration
+Configuration used to define how the deployment would be created:
+
+- mem
```

Comment:
Sorry I have not much openshift knowledge, but I would like to know what will be the use of fields like, memory, cpu etc.. after the component is created and if the user/developer has not created this file in his application and we create it based on the parameters passed to `odo create` command?

Reply:
Or do we also intend to watch over this configuration for every update of it, a new pod with the new config will be created and the old pod destroyed?

Reply:
I suspect that this configuration will be used only when the component is created.

In the future, we might add something to the `odo update` command that will modify existing components, but I think that this proposal is about new components (making it easier to replicate component creation)
Or am I wrong @jorgemoralespou?

Reply:
this is mainly about component creation.
In the future (when there is a need for it), we might add something to the `odo update` command or add the flag to `odo config` that will instruct odo to make changes to the existing components.

Reply:
@anmolbabu these are the resource limits for the pod being deployed on openshift cluster.
similar to how much amount of memory and CPU allocated to a VM.

Reply:
> @anmolbabu these are the resource limits for the application pod being deployed on openshift cluster.
> similar to how much amount of memory and CPU allocated to a VM.

@syamgk I know, I only wanted to understand as @kadel confirms that this is only for creating the components or do we also intend to re-create an already created pod if a local config was created post component creation(in which case we would need to watch for changes in expected config path) with the new config